mrp_multi_level/tests/test_mrp_multi_level.py

```diff
@@ -0,0 +1,930 @@
+            [("product_mrp_area_id.product_id", "=", self.prod_min.id)]
+        )
+        self.assertEqual(mrp_inv_min.to_procure, 50.0)
+        # maximum order quantity:
+        mrp_inv_max = self.mrp_inventory_obj.search(
+            [("product_mrp_area_id.product_id", "=", self.prod_max.id)]
+        )
+        self.assertEqual(mrp_inv_max.to_procure, 150)
+        plans = self.planned_order_obj.search([("product_id", "=", self.prod_max.id)])
+        self.assertEqual(len(plans), 2)
+        self.assertIn(100.0, plans.mapped("mrp_qty"))
+        self.assertIn(50.0, plans.mapped("mrp_qty"))
+        # quantity multiple:
+        mrp_inv_multiple = self.mrp_inventory_obj.search(
+            [("product_mrp_area_id.product_id", "=", self.prod_multiple.id)]
+        )
+        self.assertEqual(mrp_inv_multiple.to_procure, 125)
+
+    def test_08_group_demand(self):
+        """Test demand grouping functionality, `nbr_days`."""
+        pickings = self.stock_picking_obj.search(
+            [
+                ("product_id", "=", self.prod_test.id),
+                ("location_id", "=", self.sec_loc.id),
+            ]
+        )
+        self.assertEqual(len(pickings), 5)
+        moves = self.mrp_move_obj.search(
+            [
+                ("product_id", "=", self.prod_test.id),
+                ("mrp_area_id", "=", self.secondary_area.id),
+            ]
+        )
+        supply_plans = self.planned_order_obj.search(
+            [
+                ("product_id", "=", self.prod_test.id),
+                ("mrp_area_id", "=", self.secondary_area.id),
+            ]
+        )
+        moves_demand = moves.filtered(lambda m: m.mrp_type == "d")
+        self.assertEqual(len(moves_demand), 5)
+        # two groups expected:
+        # 1. days 8, 9 and 10.
+        # 2. days 20, and 22.
+        self.assertEqual(len(supply_plans), 2)
+        quantities = supply_plans.mapped("mrp_qty")
+        week_1_expected = sum(moves_demand[0:3].mapped("mrp_qty"))
+        self.assertIn(abs(week_1_expected), quantities)
+        week_2_expected = sum(moves_demand[3:].mapped("mrp_qty"))
+        self.assertIn(abs(week_2_expected), quantities)
+
+    def test_09_isolated_mrp_area_run(self):
+        """Test running MRP for just one area."""
+        self.mrp_multi_level_wiz.with_user(self.mrp_manager).create(
+            {"mrp_area_ids": [(6, 0, self.secondary_area.ids)]}
+        ).run_mrp_multi_level()
+        this = self.mrp_inventory_obj.search(
+            [("mrp_area_id", "=", self.secondary_area.id)], limit=1
+        )
+        self.assertTrue(this)
+        # Only recently exectued areas should have been created by test user:
+        self.assertEqual(this.create_uid, self.mrp_manager)
+        prev = self.mrp_inventory_obj.search(
+            [("mrp_area_id", "!=", self.secondary_area.id)], limit=1
+        )
+        self.assertNotEqual(this.create_uid, prev.create_uid)
+
+    def test_11_special_scenario_1(self):
```

Comment:
This is a bit silly, but we could change the number of the tests, it seems that there is no number 10.
```suggestion
    def test_10_special_scenario_1(self):
```

If we do this we should change them all > 11